src/CoreWCF.NetTcp/src/CoreWCF/Channels/Framing/NetMessageFramingConnectionHandler.cs

```diff
@@ -97,7 +98,22 @@ internal static UriPrefixTable<HandshakeDelegate> BuildAddressTable(IServiceProv
                         continue;
                     }
 
-                    HandshakeDelegate handshake = BuildHandshakeDelegateForDispatcher(dispatcher);
+                    IServiceDispatcher _serviceDispatcher = null;
+                    if (!(dispatcher.Binding is CustomBinding binding))
+                    {
+                        var _customBinding = new CustomBinding(dispatcher.Binding);
+                        if (_customBinding.Elements.Find<TcpTransportBindingElement>() != null)
```

Comment:
I think this whole block is fundamentally flawed. The only difference whether this is a CustomBinding already or not should be whether to wrap it in a CustomBinding or use the existing CustomBinding, but really you don't even need to do that. A CustomBinding can be created from a CustomBinding just fine. Someone could already be using a CustomBinding to modify some aspect of the behavior, like add in an extra binding element layer for logging, in which case this code would get skipped.

Reply:
check the new version, else we can discuss on Thursday sync